sentry-core/src/test/java/io/sentry/core/cache/CacheStrategyTest.kt

```diff
@@ -0,0 +1,89 @@
+package io.sentry.core.cache
+
+import com.nhaarman.mockitokotlin2.mock
+import io.sentry.core.DateUtils
+import io.sentry.core.SentryOptions
+import java.io.File
+import java.nio.file.Files
+import kotlin.test.AfterTest
+import kotlin.test.Test
+import kotlin.test.assertEquals
+import kotlin.test.assertFalse
+import kotlin.test.assertTrue
+
+class CacheStrategyTest {
+
+    private class Fixture {
+        val dir = Files.createTempDirectory("sentry-disk-cache-test").toAbsolutePath().toFile()
+        val options = SentryOptions().apply {
+            setSerializer(mock())
+        }
+
+        fun getSUT(maxSize: Int = 5): CacheStrategy {
+            return CustomCache(options, dir.absolutePath, maxSize)
+        }
+    }
+
+    private val fixture = Fixture()
+
+    @Test
+    fun `isDirectoryValid returns true if a valid directory`() {
+        val sut = fixture.getSUT()
+
+        // sanity check
+        assertTrue(fixture.dir.isDirectory)
+
+        // this test assumes that the dir. has write/read permission.
+        assertTrue(sut.isDirectoryValid)
+    }
+
+    @Test
+    fun `Sort files from the oldest to the newest`() {
+        val sut = fixture.getSUT()
+
+        val files = createTempFiles()
+        val reverseFiles = files.reversedArray()
```

Comment:
L: It's unclear that `files` is in the right order to begin with without looking into it.
What about we flip a couple explicitly to be more clear they are out of order:

```suggestion
        val reverseFiles = files.reversedArray()
        val reverseFiles[0] = files[1];
        val reverseFiles[1] = files[0];
```

Reply:
> It's unclear that files is in the right order to begin with without looking into it

I don't follow it, I created the array in the right order, and I use the extension function to reverse it, what is unclear? the impl. of the `reversedArray` do you mean?

Reply:
`createTempFIles` didn't really tell me what order they came in, so I wrote my comment

It's `L:` nonetheless so feel free to ignore

Reply:
true that, i will rename the aux. method then for something more specific